tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Logging/Serilog/DirectSubmission/Formatting/SerilogLogFormatter.cs

```diff
@@ -0,0 +1,156 @@
+using Datadog.Trace.Logging.DirectSubmission.Formatting;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
+using Datadog.Trace.Vendors.Serilog.Events;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Logging.Serilog.DirectSubmission.Formatting
+{
+    internal static class SerilogLogFormatter
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(SerilogLogFormatter));
+
+        public static void FormatLogEvent(LogFormatter logFormatter, StringBuilder sb, ILogEvent logEvent)
+        {
+            var message = logEvent.RenderMessage();
+
+            logFormatter.FormatLog(
+                sb,
+                logEvent,
+                logEvent.Timestamp.UtcDateTime,
+                message,
+                eventId: null,
+                GetLogLevelString(logEvent.Level),
+                logEvent.Exception,
+                RenderProperties);
+        }
+
+        private static string GetLogLevelString(LogEventLevelDuck logLevel) =>
+            logLevel switch
+            {
+                LogEventLevelDuck.Verbose => DirectSubmissionLogLevelExtensions.Verbose,
+                LogEventLevelDuck.Debug => DirectSubmissionLogLevelExtensions.Debug,
+                LogEventLevelDuck.Information => DirectSubmissionLogLevelExtensions.Information,
+                LogEventLevelDuck.Warning => DirectSubmissionLogLevelExtensions.Warning,
+                LogEventLevelDuck.Error => DirectSubmissionLogLevelExtensions.Error,
+                LogEventLevelDuck.Fatal => DirectSubmissionLogLevelExtensions.Fatal,
+                _ => DirectSubmissionLogLevelExtensions.Unknown,
+            };
+
+        private static LogPropertyRenderingDetails RenderProperties(JsonTextWriter writer, in ILogEvent logEvent)
+        {
+            var haveSource = false;
+            var haveService = false;
+            var haveHost = false;
+            var haveTags = false;
+            var haveEnv = false;
+            var haveVersion = false;
+            foreach (var property in logEvent.Properties)
+            {
+                var duckProperty = property.DuckCast<KeyValuePairStringStruct>();
+                var name = duckProperty.Key;
+
+                haveSource |= LogFormatter.IsSourceProperty(name);
+                haveService |= LogFormatter.IsServiceProperty(name);
+                haveHost |= LogFormatter.IsHostProperty(name);
+                haveTags |= LogFormatter.IsTagsProperty(name);
+                haveEnv |= LogFormatter.IsEnvProperty(name);
+                haveVersion |= LogFormatter.IsVersionProperty(name);
+
+                LogFormatter.WritePropertyName(writer, name);
+                FormatLogEventPropertyValue(writer, duckProperty.Value);
+            }
+
+            var renderingDetails = new LogPropertyRenderingDetails(haveSource, haveService, haveHost, haveTags, haveEnv, haveVersion, messageTemplate: logEvent.MessageTemplate.Text);
```

Comment:
[nit] Named params would be a plus even though we call the right ones.

Reply:
Agreed, done in https://github.com/DataDog/dd-trace-dotnet/pull/2240/commits/e5ad6f6df18fc4d2d1cbd6a4d13763a3f6991360